src/logger_test.go

```diff
@@ -9,50 +9,72 @@ import (
-	// Create a temporary directory for the log folder
-	logFolderName := "log-test"
-	logFileName := "log-file"
-	defer os.RemoveAll(logFolderName)
-
-	// Test case 1: YGG_LOG_LEVEL doesn't exist, info level should be set to info
-	setupLogger(logFolderName, logFileName)
-	level := log.CurrentLevel()
-	if log.CurrentLevel() != log.LevelInfo {
-		t.Errorf("Incorrect log level. Expected: %v, Got: %v", log.LevelInfo, level)
-	}
-	// Test case 2: Unparsable level in env variable
-	os.Setenv("YGG_LOG_LEVEL", "....")
-	setupLogger(logFolderName, logFileName)
-	level = log.CurrentLevel()
-	if log.CurrentLevel() != log.LevelInfo {
-		t.Errorf("Incorrect log level. Expected: %v, Got: %v", log.LevelInfo, level)
+	testCases := []struct {
+		name          string
+		logLevelEnv   string
+		expectedLevel log.Level
+	}{
+		{
+			name:          "YGG_LOG_LEVEL doesn't exist, info level should be set to info",
+			logLevelEnv:   "",
+			expectedLevel: log.LevelInfo,
+		},
+		{
+			name:          "Unparsable level in env variable",
+			logLevelEnv:   "....",
+			expectedLevel: log.LevelInfo,
+		},
+		{
+			name:          "Everything set up correctly",
+			logLevelEnv:   "debug",
+			expectedLevel: log.LevelDebug,
+		},
 	}
 
-	// Test case 3: Everything set up correctly
-	os.Setenv("YGG_LOG_LEVEL", "debug")
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			// Create a temporary directory for the log folder
+			logFolderName := "log-test"
+			logFileName := "log-file"
+			defer os.RemoveAll(logFolderName)
 
-	logfile := setupLogger(logFolderName, logFileName)
-	if _, err := os.Stat(logFolderName); os.IsNotExist(err) {
-		t.Errorf("Log folder not created: %v", err)
-	}
-	logFilePath := filepath.Join(logFolderName, logFileName)
-	if _, err := os.Stat(logFilePath); os.IsNotExist(err) {
-		t.Errorf("Log file not created: %v", err)
-	}
-	level = log.CurrentLevel()
-	if level != log.LevelDebug {
-		t.Errorf("Incorrect log level. Expected: %v, Got: %v", log.LevelDebug, level)
-	}
-	flags := log.Flags()
-	expectedFlags := log.Lshortfile | log.LstdFlags
-	if flags != expectedFlags {
-		t.Errorf("Incorrect log flags. Expected: %v, Got: %v", expectedFlags, flags)
-	}
+			// Set YGG_LOG_LEVEL environment variable
+			if tc.logLevelEnv != "" {
+				os.Setenv("YGG_LOG_LEVEL", tc.logLevelEnv)
+			}
+
+			logfile := setupLogger(logFolderName, logFileName)
+
+			// Verify log level
+			level := log.CurrentLevel()
+			if level != tc.expectedLevel {
+				t.Errorf("Incorrect log level. Expected: %v, Got: %v", tc.expectedLevel, level)
+			}
+
+			// Verify log file and folder
+			if _, err := os.Stat(logFolderName); os.IsNotExist(err) {
+				t.Errorf("Log folder not created: %v", err)
+			}
+			logFilePath := filepath.Join(logFolderName, logFileName)
+			if _, err := os.Stat(logFilePath); os.IsNotExist(err) {
+				t.Errorf("Log file not created: %v", err)
+			}
+
+			// Verify log flags
+			flags := log.Flags()
+			expectedFlags := log.Lshortfile | log.LstdFlags
```

Comment:
What about moving those to the table as well?

Reply:
moved